poetry/utils/helpers.py

```diff
@@ -100,3 +101,21 @@ def _on_rm_error(func, path, exc_info):
 
 def safe_rmtree(path):
     shutil.rmtree(path, onerror=_on_rm_error)
+
+
+def parse_author(address):  # type: (str) -> tuple
+    """Parse name and address parts from an email address string.
+
+    .. note::
+
+       If the input string does not contain an `@` character, it is
+       assumed that it represents only a name without an email address.
+
+    :param address: the email address string to parse.
+    :return: a 2-tuple with the parsed name and email address.  If a
+             part is missing, ``None`` will be returned in its place.
+    """
+    if "@" not in address:
```

Comment:
Considering comma-separated inputs( In [15]: ), please put this below `parseaddr()` if not raising exception.
https://github.com/sdispater/poetry/pull/981#pullrequestreview-218088417

```suggestion
    name, email = parseaddr(address)
    if not name and "@" not in address:
        return (email, None)
```

Reply:
If I’m doing this, then just providing a full name without an email address does not work any more, as evidenced by the failure of the existing `test_package_authors()` test as well as one of the test cases that are part of this PR:

```
    def test_package_authors():
        package = Package("foo", "0.1.0")

        package.authors.append("Sébastien Eustace <sebastien@eustace.io>")
        assert package.author_name == "Sébastien Eustace"
        assert package.author_email == "sebastien@eustace.io"

        package.authors.insert(0, "John Doe")
>       assert package.author_name == "John Doe"
E       AssertionError: assert 'John' == 'John Doe'
E         - John
E         + John Doe

tests/packages/test_package.py:15: AssertionError

[...]

    def test_parse_author_simple_name_only():
        name, email = parse_author("John Doe")
>       assert name == "John Doe"
E       AssertionError: assert 'John' == 'John Doe'
E         - John
E         + John Doe

tests/utils/test_helpers.py:83: AssertionError
```

This is due to the fact that Python’s `parseaddr()` does the following for `John Doe`:

```Python console
>>> parseaddr('John Doe')
('', 'John')
```

Hence I went the pragmatic route and simply checked for the presence of an “@” character in the provided input string, as I would assume that otherwise the user’s intention was to specify a name only.  (Apart from that, I believe—at least for Poetry’s purposes—that any author email address should always contain an “@” character.)

Having a comma character in the input string might happen of course.  I’d consider this a rather quirky edge case, though, so I think the benefits of this PR outweigh the drawbacks.

If this really is a blocker for this PR to get merged, it would be great if you could specify what `parse_author()` is supposed to return for different kinds of input strings that contain a comma.

Reply:
I see. Thanks for illustration. Now it only needs check `'@'` twice in email part, before and after `parsesddr()` to cover this two cases.

And `str(None) == 'None'` is also a concern for who uses this as the input to database.
So I followed the empty string style same as `parseaddr()`

Reply:
> And `str(None) == 'None'` is also a concern for who uses this as the input to database.
> So I followed the empty string style same as `parseaddr()`

I am well aware that `str(None)` is not always useful.  Initially, I actually had `parse_author()` return empty strings for missing fields.  However, the already existing `test.packages.test_package.test_package_authors()` test contains the following (note the final assertion):

```python
def test_package_authors():
    package = Package("foo", "0.1.0")

    [...]

    package.authors.insert(0, "John Doe")
    assert package.author_name == "John Doe"
    assert package.author_email is None
```

Therefore, I assumed that this was the expected way to go for Poetry, and I changed my implementation to returning `None`.

If this basic behavior should be changed to using empty strings instead of `None`, it would probably require a thorough evaluation first on how far reaching consequences this would entail.  However, I believe that this is out of scope for this PR and should be handled separately if so desired.  After all, I only stumbled upon this issue because of having a fancy unicode character in my company’s name that Poetry was unable to deal with. ;-)

Reply:
It can be another issue.

Reply:
> I see. Thanks for illustration. Now it only needs check `'@'` twice in email part, before and after `parsesddr()` to cover this two cases.

Okay—just to be on the same page—so this is how you envision `parse_author()` to handle the funky use cases?

```python console
>>> parse_author("asf,dfu@t.b")
(None, 'asf')
>>> parse_author("asf,<dfu@t.b>")
(None, 'asf')
>>> parse_author("asf, dfu@t.b")
(None, 'asf')
```

Reply:
After a quick search in this project, what matters the most is this condition statement: https://github.com/sdispater/poetry/blob/2f2cec03b6b14e882522f7f6042ecdafebda2ca5/poetry/layouts/layout.py#L71


Though the author name is usually initialized as `None`.

I will open another issue and PR after a deeper investigation

Reply:
> > I see. Thanks for illustration. Now it only needs check `'@'` twice in email part, before and after `parsesddr()` to cover this two cases.
> 
> Okay—just to be on the same page—so this is how you envision `parse_author()` to handle the funky use cases?
> 

What I expected is:

 ```python
>>> parse_author("asf,dfu@t.b")
('asf'. '')
>>> parse_author("asf,<dfu@t.b>")
('asf', '')
>>> parse_author("asf, dfu@t.b")
('asf', '')
```

`parse_author()` will place those obviously non-email information into author name part.
Though it loses the real email.

But sometimes it doesn't really need full information.

And considering the impact including in breaking tests and further discussion, I thought this PR is better to be like this to solve the issue immediately.

Reply:
Thank you for elaborating.  Please see commit edba8fd that implements this extended behavior!